lib/livebook/session.ex

```diff
@@ -814,6 +833,20 @@ defmodule Livebook.Session do
     {:noreply, handle_save_finished(state, result, file, default?)}
   end
 
+  def handle_info(:system_memory, state) do
+    {:noreply, state |> system_memory_usage() |> schedule_system_memory()}
+  end
+
+  def handle_info({:memory_usage, memory_usage}, state) do
+    Process.cancel_timer(state.system_memory_timer_ref)
```

Comment:
I think no. Here we are canceling the system memory tracking for when we don't have a runtime. Since we have a runtime, we don't need this anymore until a disconnection happens.

Reply:
Gotcha, good point. So we need to schedule it back when we set runtime to `nil` then :)